tests/test_frequencies.py

```diff
@@ -138,6 +138,51 @@ def test_get_pivots_by_invalid_unit():
     with pytest.raises(ValueError, match=r".*invalid_unit.*is not supported.*"):
         pivots = get_pivots(observations=[], pivot_interval=1, start_date=2015.0, end_date=2016.0, pivot_interval_units="invalid_unit")
 
+
+@pytest.mark.parametrize(
+    "start, end, expected_pivots",
+    [
+        (
+            "2022-01-01",
+            "2022-04-01",
+            ("2022-01-01", "2022-02-01", "2022-03-01", "2022-04-01")
+        ),
+        (
+            "2022-01-31",
+            "2022-03-31",
+            ("2022-01-31", "2022-02-28", "2022-03-31")
+        ),
+        # Note that Jan 31 to Apr 30 gives the same amount of pivot points as
+        # Jan 31 to Mar 31.
+        (
+            "2022-01-31",
+            "2022-04-30",
+            ("2022-02-28", "2022-03-30", "2022-04-30")
+        ),
+        # However, in practice, the interval is more likely to be Jan 30 to Apr
+        # 30 as long as the start date is calculated relative to the end date
+        # (i.e. start date = 3 months before Apr 30 = Jan 30).
+        # That interval includes an additional pivot point as expected.
```

Comment:
I love how clear and well-documented these tests are. They make the original issue clear to me even a month later, so hopefully it will still be clear to other future developers for the same reasons.

Reply:
Love to hear that!